tinygrad/renderer/cstyle.py

```diff
@@ -35,7 +35,8 @@ class CStyleLanguage(NamedTuple):
     UnaryOps.SIN: lambda x: f"sin({x})",
     UnaryOps.SQRT: lambda x: f"sqrt({x})",
     BinaryOps.ADD: lambda a,b: f"({a}+{b})", BinaryOps.SUB: lambda a,b: f"({a}-{b})",
-    BinaryOps.MUL: lambda a,b: f"({a}*{b})", BinaryOps.DIV: lambda a,b: f"({a}/{b})",
+    BinaryOps.MUL: lambda a,b: f"({a}*{b})" if "INFINITY" not in [a,b] else (f"(({a}>=0?1:-1)*({b}>=0?1:-1)*INFINITY)"),
```

Comment:
Why is this needed?

Reply:
I thought that is a problem when multiplying values. But I'm wrong, diskcache uses the old program code.
So, using `options.setFastMathEnabled_(False)` is enough
I updated a new commit.